### DISCUSSION TITLE
imgui with the SDL2 renderer (not gl/dx...)

### DISCUSSION THREAD

**Discussion by ucacaxm** [ocornut/imgui#4569] [New Users Build/Link/Run issues ONLY!]
Hi,

Can I use imgui with the basic SDL2 renderer: the one create with SDL_CreateRenderer ?
(without using OpenGL or DX or any other renderer).

Best,

**Comment by ocornut** [comment]
The current version of SDL_Renderer API makes it impossible to render textured triangles (it's a very poor API ihmo and I personally think you should avoid using SDL_Renderer. You can use the OpenGL/DX9/DX11 backend with SDL).

But next version of SDL will upgrade SDL_Renderer to allow rendering textured triangles and when that is released we will add the backend for it. There's a yet unmerged pull request doing just that: https://github.com/ocornut/imgui/pull/3926

**Reply by ocornut** [reply]
According to https://libsdl.org/next.php the next version of SDL will be released in 52 days and should support it.

**Comment by ucacaxm** [comment]
Thank you for the answer.

I currently have trouble mixing imgui/SDL/GL and the renderer: I manage to get one or the other, but not both.

I'm going to get through this by opening two windows: one with imgui/SDL/GL and one with the SDL/renderer.
For what I want it will be ok.

best,

**Comment by ucacaxm** [comment]
Thank to this 
https://github.com/ocornut/imgui/issues/4264
I manage to mix imgui/OpenGL3 and SDL-Renderer 

Thanks
